packages/flutter/lib/src/material/text_selection.dart

```diff
@@ -20,64 +23,390 @@ const double _kHandleSize = 22.0;
+        });
+
+        final RenderBox renderBoxMoreButton = _moreButtonKey.currentContext.findRenderObject() as RenderBox;
+        final double itemsWidth = renderBoxContainer.size.width - remainingContainerWidth;
+
+        _menuContentWidth = _shouldShowMoreButton
+          ? itemsWidth + renderBoxMoreButton.size.width
+          : itemsWidth;
+      });
+    });
+  }
+
   @override
   Widget build(BuildContext context) {
     final MaterialLocalizations localizations = MaterialLocalizations.of(context);
-    final List<Widget> items = <Widget>[
-      if (handleCut != null) FlatButton(child: Text(localizations.cutButtonLabel), onPressed: handleCut),
-      if (handleCopy != null) FlatButton(child: Text(localizations.copyButtonLabel), onPressed: handleCopy),
-      if (handlePaste != null) FlatButton(child: Text(localizations.pasteButtonLabel), onPressed: handlePaste),
-      if (handleSelectAll != null) FlatButton(child: Text(localizations.selectAllButtonLabel), onPressed: handleSelectAll),
-    ];
+
+    final List<Widget> items = <Widget>[];
+    _itemKeys.removeRange(0, _itemKeys.length);
+    if (widget.handleCut != null) {
+      _itemKeys.add(GlobalKey());
+      items.add(FlatButton(
+        key: _itemKeys[_itemKeys.length - 1],
+        child: Text(localizations.cutButtonLabel),
+        onPressed: () {
+          setState(() {
+            _overflowOpen = false;
+          });
+          widget.handleCut();
+        },
+      ));
+    }
+    if (widget.handleCopy != null) {
+      _itemKeys.add(GlobalKey());
+      items.add(FlatButton(
+        key: _itemKeys[_itemKeys.length - 1],
+        child: Text(localizations.copyButtonLabel),
+        onPressed: () {
+          setState(() {
+            _overflowOpen = false;
+          });
+          widget.handleCopy();
+        },
+      ));
+    }
+    if (widget.handlePaste != null) {
+      _itemKeys.add(GlobalKey());
+      items.add(FlatButton(
+        key: _itemKeys[_itemKeys.length - 1],
+        child: Text(localizations.pasteButtonLabel),
+        onPressed: () {
+          setState(() {
+            _overflowOpen = false;
+          });
+          widget.handlePaste();
+        },
+      ));
+    }
+    if (widget.handleSelectAll != null) {
+      _itemKeys.add(GlobalKey());
+      items.add(FlatButton(
+        key: _itemKeys[_itemKeys.length - 1],
+        child: Text(localizations.selectAllButtonLabel),
+        onPressed: () {
+          setState(() {
+            _overflowOpen = false;
+          });
+          widget.handleSelectAll();
+        },
+      ));
+    }
 
     // If there is no option available, build an empty widget.
     if (items.isEmpty) {
       return Container(width: 0.0, height: 0.0);
     }
 
-    return Material(
-      elevation: 1.0,
-      child: Container(
-        height: _kToolbarHeight,
-        child: Row(mainAxisSize: MainAxisSize.min, children: items),
+    // If _itemsInFirstMenu hasn't been calculated yet, render offstage for one
```

Comment:
On the face of it, it looks like "If _menuContentWidth" hasn't been calculated yet.

Reply:
This code was removed with the RenderBox approach.